src/fpPractice.test.js

```diff
@@ -0,0 +1,349 @@
+test("Test: FP Assignment Quesions", () => {
+
+    //1st Assingmnet question
+    function fibonacci(n) {
+        const fibs = [0, 1]
+        function fibInternal(n) {
+            if (fibs[n]) {
+                return fibs[n]
+            } else {
+                for (let i = 2; i <= n; i++) {
+                    fibs[i] = fibs[i - 1] + fibs[i - 2]
+                }
+                return fibs[n]
+            }
+        }
+        return fibInternal(n)
+    }
+
+    function indexToFibValue(arr) {
+        return arr.map((index) => fibonacci(index))
+    }
+
+
+    expect(indexToFibValue([2, 1, 5, 7])).toEqual([1, 1, 5, 13])
+    expect(indexToFibValue([1, 2, 3, 4])).toEqual([1, 1, 2, 3])
+
+
+    //2nd Assignment Question
+
+    function convertAvgTempToFar(temp) {
+        return parseFloat(((temp * (9 / 5)) + 32).toFixed(2))
+    }
+
+    function averageTemperatureToFahrenheit(ArrOfobj) {
+        return ArrOfobj.map((obj) => {
+            let fahrenheit = convertAvgTempToFar(obj.averageTemperature)
```

Comment:
Always create a shallow copy. Otherwise the original array will also get modified.

Reply:
Thank you for the feedback. I'll make sure to create a shallow copy to avoid modifying the original array.